lms/djangoapps/django_comment_client/permissions.py

```diff
@@ -31,19 +31,14 @@ def has_permission(user, permission, course_id=None):
 CONDITIONS = ['is_open', 'is_author', 'is_question_author', 'is_team_member_if_applicable']
 
 
+@request_cached
 def get_team(commentable_id):
     """ Returns the team that the commentable_id belongs to if it exists. Returns None otherwise. """
```

Comment:
So, it wasn't that, but something similar: it was grabbing the value for a given cache key out of the cache, and checking it in a truthy manner.  For `None` results, this failed the check and reran the underlying wrapped function again and again.

I changed this to check if the cache key exists in the cache, and if it does, to simply return the result of getting that value from cache, no matter what it is.  This has, appropriately, reduced the query count compared to the base, saving two queries (30 vs 32) for the particular test case I had originally modified.